cpp/ps0008/ps150days.cpp

```diff
@@ -53,110 +46,53 @@ string userInputKey() {
 
-    /* implement your function and consider that:
-        - all uppercase letters in ciphertext are uppercase letters in plaintext
-        - all lowercase letters in ciphertext are lowercase letters in plaintext
-        - all digits, special characters, spaces in ciphertext are the same in plaintext
-    */
-    if (key == 0) {
-        cout << msg;
-        return;
+    cout << "plaintext:  ";
+    
+    if (abs(key) >= 26) {
+        key %= 26; // no need for loop
+                   // 26 steps give the same point
     }
 
-    else if (abs(key) > 26) {
-
-        while (abs(key) >= 26) {
-            key %= 26;
-        }
-
-       
-
-       
-
-    }
-
-
-   
-
+    if (key == 0) {
+        cout << plainText;
+        return;
+    } // key may be multiple of 26
 
-   
+    int i = 0;
     for (char ch : msg) {
 
         if (isalpha(ch)) {
 
             //decrypt then add to plainText
             int t = int(ch);
             t -= key;
-            cout << "t = " << t <<"\n";
-            if ( (islower(ch) && t >= 97 && t <= 122)   || isupper(ch) && t>=65 && t<=90 ){
-                ch = static_cast<char>(t);
-                cout << "ch = " << ch << endl;
-                plainText += ch;
-                cout << "plainText = " << plainText << endl;
-            }
 
-            else if (islower(ch) && t < 97) {
-                t += 26;
-                ch = static_cast<char>(t);
-                cout << "ch = " << ch << endl;
-                plainText += ch;
-                cout << "plainText = " << plainText << endl;
-            }
-            else if (islower(ch) && t > 122) {
-                t -= 26;
-                ch = static_cast<char>(t);
-                cout << "ch = " << ch << endl;
-                plainText += ch;
-                cout << "plainText = " << plainText << endl;
-            }
-            else if (isupper(ch) && t < 65) {
+            if ((islower(ch) && t < 97) || (isupper(ch) && t < 65)) {
                 t += 26;
-                ch = static_cast<char>(t);
-                cout << "ch = " << ch << endl;
-                plainText += ch;
-                cout << "plainText = " << plainText << endl;
             }
-            else if (isupper(ch) && t > 90) {
+            else if ((islower(ch) && t > 122) || (isupper(ch) && t > 90)) {
                 t -= 26;
-                ch = static_cast<char>(t);
-                cout << "ch = " << ch << endl;
-                plainText += ch;
-                cout << "plainText = " << plainText << endl;
             }
             
-          
-        }
-
-
-        else {
-            plainText += ch;
-            cout << "plainText == " << plainText << endl;
+            ch = t;
         }
-
-
+        
+        // the previous lines line of code were repeated, DRY! 
+        plainText[i++] = ch;
     }
 
-    cout << plainText;
-
-  
-
-}
\ No newline at end of file
+    cout << plainText; // no need to repeat for every single character
+}

```

Comment:
- [x] defined function takes two parameters (ciphertext and key)for decryption process
- [x] considered the case of the letter in the message while decryption
- [x] skip numbers and other special characters while decryption
- [x] considered that the key may be positive or negative value
- [x] returned the actual message (message before encryption)